ocaml/idl/latex_backend.ml

```diff
@@ -123,253 +83,206 @@ let flatten stuff =
   f "" stuff
 
 let string_of_qualifier = function
-  | StaticRO   -> "$\\mathit{RO}_\\mathit{ins}$"
-  | DynamicRO  -> "$\\mathit{RO}_\\mathit{run}$"
-  | RW         -> "$\\mathit{RW}$"
+  | StaticRO   -> "_RO~ins~_"
+  | DynamicRO  -> "_RO~run~_"
+  | RW         -> "_RW_"
 
 
 let string_of_open_product release =
-  let xe = if release.opensource = [] then "XE" else "" in
-  let dep = if release.internal_deprecated_since = None then "" else "(deprecated)" in
-  Printf.sprintf "%s %s &" xe dep
-
-let of_enum_alias name options = [
-  "\\begin{longtable}{|ll|}";
-  "\\hline";
-  "{\\tt enum " ^ (escape name) ^ "} & \\\\";
-  "\\hline" ] @
-  (List.map (fun (option, description) ->
-       hgap ^ "{\\tt " ^ (escape option) ^ "} & " ^ (escape description) ^ " \\\\") options) @
-  [
-    "\\hline";
-    "\\end{longtable}"
-  ]
-
-
-let of_content x closed =
-  let rec f prefix = function
-    | Field{release=release; qualifier=qualifier; field_name=name; ty=ty; field_description=description} ->
-      [ sprintf "%s%s & %s {\\tt %s} & %s & %s \\\\"
-          (if closed then
-             string_of_open_product release
-           else
-             "")
-          (string_of_qualifier qualifier)
-          prefix (escape name) (of_ty ty) (escape description) ]
-    | Namespace(_, fields) -> List.concat (List.map (f prefix) fields)
-  in f "" x
+  if release.internal_deprecated_since = None then "" else "**Deprecated.** "
 
-(*
-  let header = [ "\\documentclass[8pt]{article}";
-  "\\usepackage{geometry}";
-  "\\usepackage{layout}";
-  "\\geometry{";
-  "\left=2.0cm,";
-  "\right=2.5cm,";
-  "\top=3.5cm,";
-  "\bottom=3cm";
-  "}";
-  "\\usepackage{graphics}";
-  "\\usepackage{longtable}";
-(* "\\usepackage[a3paper,pdftex]{geometry}"; *)
-  "\\usepackage{fancyhdr}";
-  "\\setlength\\parindent{0pt}";
-  "\\begin{document}" ]
-  let footer = [ "\\end{document}" ]
-*)
-
-
-(* Output API parameter table entry *)
-let mk_latex_param p =
-  String.concat " "
-    ["{\\tt"; of_ty p.param_type; "}"; "&";
-     escape p.param_name; "&";
-     escape p.param_doc; "\\\\ \\hline"; "\n"]
-
-let mk_latex_error err =
-  sprintf "{\\tt %s}" (escape err.err_name)
-
-let space = "\\vspace{0.3cm}"
+let print_field_content printer x =
+  let rec f = function
+    | Field{release=release; qualifier=qualifier; field_name=name; ty=ty; field_description=description} ->
```

Comment:
We can make this a bit shorter using something called ["field punning"](https://realworldocaml.org/v1/en/html/records.html#field-punning):
```
| Field{release; qualifier; field_name=name; ty; field_description=description} ->
```
When the name of the record field has the same as the variable that we want to bind it to, we can shorten the code this way. (The same applies when constructing records.)